src/templates.ts

```diff
@@ -110,9 +111,35 @@ export function getTemplates(options: ModuleOptions, uiConfig: Record<string, an
   writeThemeTemplate(theme)
 
   async function generateSources() {
-    let sources = ''
+    if (!nuxt) {
+      return '@source "./ui";'
+    }
+
+    const sources: string[] = []
 
-    if (!!nuxt && !!resolve && options.experimental?.componentDetection) {
+    // Add layer sources
+    for (const layer of nuxt.options._layers || []) {
+      const srcDir = layer.config.srcDir || layer.cwd
+      const relativePath = relative(nuxt.options.buildDir, srcDir)
+
+      sources.push(`@source ${JSON.stringify(`${relativePath}/**/*.{vue,ts,tsx,js,jsx,mjs,mts}`)};`)
```

Comment:
We could use the configured extensions instead of the hardcoded ones: https://nuxt.com/docs/4.x/api/nuxt-config#extensions. I am not sure how/if they are applied to layers.